enhancements/network/allow-mtu-and-overlay-port-changes.md

```diff
@@ -0,0 +1,165 @@
+
+* Allow to change MTU configuration in te underlay interfaces.
+
+* Allow to change MTU, and either port in a single operation.
+
+* Other safe or unsafe configuration changes.
+
+## Proposal
+
+When the cluster network operator detects the change it will deploy a new
+deaemonset with restartPolicy Mever which is responsible for doing the changes.
+
+Even though both changing the MTU, and changing the port have a lot in common,
+there are some key differences and therefore the daemonset will be slightly
+different.
+
+When decreasing the MTU each of the deamonset will:
+
+1. Verify that the new MTU is valid for that node. If it is valid add to
+   itself the annotation `openshift.io/cno-able-to-change-mtu`.
+2. Wait until the timestamp defined in `CHANGE_MTU_INIT_TS`
+3. Verify that every pod of that daemonset is running and has the annotation
+   `openshift.io/cno-able-to-change-mtu` if any of the pods is not running, or
+   is missing that annotation the pod will abort the procedure.
+4. Cordon the node where its running.
+5. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+6. Proceed to change the MTU of `eth0` on each one of the pods netnamespace.
+7. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+8. Proceed to change the MTU of each pod in the virtual switch.
+9. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+10. In the case of OVN Kubernetes proceed to change the MTU of the interfaces
+   ovn-k8s-mp0, br-local, and ovn-k8s-gw0 sequentially without any delay in
+   between.
+
+   In the case of OpenShiftSDN proceed to change the MTU of the interfaces
+   vxlansys.
+
+13. Uncordon the node
+
+14. Annotate itself with `openshift.io/cno-succesuflly-changed-mtu` and exit
+    with code 0.
+
+When increasing the MTU the same process applies, except steps 10, 8, and 6 are done in reverse order.
+
+Changing the overlay ports is a similar procedure, we will depoy a daemonset
+where each pod will:
+
+1. Verify that the port is not being used at the moment by anything else and
+   add to itself the annotation `openshift.io/cno-able-to-change-port`
+2. Wait until the timestamp defined in `CHANGE_PORT_INIT_TS`
+3. Verify that every pod of that daemonset is running and has the annotation
+   `openshift.io/cno-able-to-change-port` if any of the pods is not running, or
+   is missing that annotation the pod will abort the procedure.
+4. Cordon the node where its running.
+5. Wait a grace period. This grace period is based on
+   `CHANGE_PORT_INIT_TS` rather than the time of finalization of the
+   previous step.
+6. Actually change the port:
+   For OpenShift SDN we'll delete the port:
+
+   `ovs-vsctl --if-exists del-port br0 vxlan0`
+
+   And create it again with no delay at all:
```

Comment:
Is this procedure actually already tested or is this just a guess/placeholder? If the latter, you don't need to specify it in that much detail here.

Reply:
I tested this manually. Should I make it less detailed anyway?

Reply:
@juanluisvaladas I'd like to leave detailed steps where possible. Could be helpful for QE to understand the flow of the things. Right now it seems useful to understand what's being planned.